### DISCUSSION TITLE
endpoints just return bundle file

### DISCUSSION THREAD

**Discussion by leookun** [withastro/roadmap#955] [Proposal]
### Body

# Summary

Can we add new route named "bundle files route" beyond html content or api endpoints ? 
The "bundle files route" responds to pure javascript content or css content

# Background & Motivation

I'm using Astro to create an open platform that provides users with apis、 pages、apps and javascript SDKS. When I tried to create javascript SDKS, I found that I had to write  build code  which is overlaps with astro, and my source code is all from apps created by Astro

# Goals

- bundle files route
- respond javascript or css content 

# Example
For example, the following file named sdk-entry.tsx , path is pages/perview/sdk-entry.tsx

<img width="1349" alt="image" src="https://github.com/withastro/roadmap/assets/131544788/d7bab080-b615-4c8b-b910-1374cd4883b6">

when user access by script tag in his html :

```html
<script src="http://explame.com/perview/sdk-entry" />
```
He will get the bundle .

My English is not good, sorry

**Comment by Fryuni** [comment]
You can already do this:
```html
<script src="./path/to/local/file.ts" />
```

An endpoint will be generated with the bundle of that file.
